### PR TITLE
[Merged by Bors] - feat: SmartModule Lookback feature

### PR DESCRIPTION
**Pull request by galibey**
Added a section about Lookback feature.

### CODE REVIEW

File: content/smartmodules/lookback.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    ...
+}
+```
+- `lookback` parameter is specified in transform configuration. It defines how many lookback records SmartModule receives:
+```yaml
+transforms:
+  - uses: local/filter-with-lookback@0.1.0
+    lookback:
+      last: 10 # we want last 10 records
+```
```
Comment (ajhunyady) [inline content/smartmodules/lookback.md:23]: I thought we also built lookback based on `time`.

File: content/smartmodules/lookback.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+---
+If Fluvio topic is empty, `look_back` is never called.
+
+-> If you start processing not from the end, you may receive records both as lookback record and as regular processing
```
Comment (ajhunyady) [inline content/smartmodules/lookback.md:27]: If you start processing from an offset other than the end, you will...

File: content/smartmodules/lookback.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+-> Lookback is only supported for SmartModules that are run on SPU
```
[the remark this answers is not in the document]
Comment (ajhunyady) [reply content/smartmodules/lookback.md:29]: If connectors leverage the SPU, we should update the statement.

File: content/smartmodules/lookback.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
+---
+title: Lookback
+---
+
+The Lookback feature in Fluvio SmartModule is the way to access records that existed before the SmartModule started allowing it to build its internal state depending on what the topic currently has.
```
[the remark this answers is not in the document]
Comment (galibey) [reply content/smartmodules/lookback.md]: Changed

File: content/smartmodules/lookback.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (galibey) [reply content/smartmodules/lookback.md:29]: for source connectors it is not going to work in the current version

> If connectors leverage the SPU, we should update the statement.

Only sink connectors uses SPU-side SmartEngine

[the remark this answers is not in the document]
Comment (ajhunyady) [reply content/smartmodules/lookback.md:29]: Let's update the doc please.

[the remark this answers is not in the document]
Comment (galibey) [reply content/smartmodules/lookback.md]: Added a sentence that Source Connectors don't support it.

**Pull request closed without merging**